packages/fal/src/fal-transcription-model.ts

```diff
@@ -0,0 +1,380 @@
+import {
+  TranscriptionModelV1,
+  TranscriptionModelV1CallWarning,
+} from '@ai-sdk/provider';
+import {
+  combineHeaders,
+  convertUint8ArrayToBase64,
+  createJsonErrorResponseHandler,
+  createJsonResponseHandler,
+  getFromApi,
+  parseProviderOptions,
+  postJsonToApi,
+} from '@ai-sdk/provider-utils';
+import { z } from 'zod';
+import { FalConfig } from './fal-config';
+import { falErrorDataSchema, falFailedResponseHandler } from './fal-error';
+import { FalTranscriptionModelId } from './fal-transcription-settings';
+import { FalTranscriptionAPITypes } from './fal-api-types';
+
+// https://fal.ai/models/fal-ai/whisper/api?platform=http
+const falProviderOptionsSchema = z.object({
+  /**
+   * Language of the audio file. If set to null, the language will be automatically detected. Defaults to null.
+   *
+   * If translate is selected as the task, the audio will be translated to English, regardless of the language selected.
+   */
+  language: z
+    .enum([
```

Comment:
how likely is this to change? it's some code so we could also change to string and rely on the API for detailed checks

Reply:
maybe we could do something clever like we do for the model ids which is allow `string & {}` not sure how  you do that in zod, but I think it would make sense

Reply:
![CleanShot 2025-04-12 at 19 07 22@2x](https://github.com/user-attachments/assets/b1f71568-1d94-4d4a-9658-55b509359696)

Reply:
lets consider that when/if we pull out language into a top-level option